plugins/linux-v4l2/CMakeLists.txt

```diff
@@ -1,31 +1,43 @@
-project(linux-v4l2)
+cmake_minimum_required(VERSION 3.22...3.25)
+
+legacy_check()
 
 option(ENABLE_V4L2 "Build OBS with v4l2 support" ON)
+option(ENABLE_UDEV "Build linux-v4l2 with UDEV support" ON)
+
 if(NOT ENABLE_V4L2)
-  obs_status(DISABLED "linux-v4l2")
+  target_disable(linux-v4l2)
   return()
 endif()
 
-option(ENABLE_UDEV "Build linux-v4l2 with UDEV support" ON)
-
 find_package(Libv4l2 REQUIRED)
 find_package(FFmpeg REQUIRED COMPONENTS avcodec avutil avformat)
 
+if(OS_FREEBSD OR OS_OPENBSD)
+  set(CMAKE_REQUIRED_INCLUDES "/usr/local/include")
+endif()
+include(CheckCSourceCompiles)
+set(V4L_TEST_SOURCE "#include<linux/videodev2.h>\nint main(){return 0;}")
+check_c_source_compiles("${V4L_TEST_SOURCE}" HAVE_VIDEODEV2_HEADER)
+
+if(NOT HAVE_VIDEODEV2_HEADER)
+  message(FATAL_ERROR "Required system header <linux/videodev2.h> not found.")
+endif()
```

Comment:
Maybe, the main purpose was to use a _single_ pattern for this kind of check throughout the code base, regardless of whether just a header, a header with an associated library, or a header with an _optional_ associated library needs to be checked.

Reply:
Hmm. I'd sort of prefer to use the tools that CMake gives us instead of having to use a lot of boilerplate for simple include checks, but we can perhaps explore that in the future. They do appear to be functionally equivalent in that `CHECK_INCLUDE_FILE` places the specified include name into an `#include <include_name>` in a similar C source file. It does also have options for requiring link options and libraries. The main difference for us is that we would have less CMake to write/maintain:
```cmake
include(CheckIncludeFile)
CHECK_INCLUDE_FILE("linux/videodev2.h" HAVE_VIDEODEV2_HEADER)
```

As I said earlier though, we can explore that as a later change if we're so inclined.

Reply:
We'd still need to adjust the include path for FreeBSD, but the check should support that as well.